libnavigation-core/src/test/java/com/mapbox/navigation/core/internal/utils/RoutesUpdateReasonHelperParameterizedTest.kt

```diff
@@ -0,0 +1,274 @@
+package com.mapbox.navigation.core.internal.utils
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.extensions.applyDefaultNavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesExtra
+import com.mapbox.navigation.core.trip.session.OffRouteObserver
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class RoutesUpdateReasonHelperParameterizedTest(
```

Comment:
Let's also add a test for multi-leg routes - what if we reroute after already on a second leg for example.